### PR TITLE
feature: Discord OAuth2 Provider

### PR DESCRIPTION
**Pull request by mgkbadola**
Added login support for Discord. If this is approved and merged, I would like to add ~~it's~~ its support to Remark42 (love that service).

A request: Since I am currently participating in Hacktoberfest, if this PR is to be merged, kindly add the label `hacktoberfest-accepted` to it. Many thanks!

### CODE REVIEW

**Comment by umputun** [comment]
Thanks. I don't see anything wrong with this PR. @paskal - can you please take a look and double-check?

**Comment by umputun** [comment]
@mgkbadola - pls fix the linter issue, look like a public method without a comment

**Comment by mgkbadola** [comment]
> @mgkbadola - pls fix the linter issue, look like a public method without a comment

on it 👍🏽

**Comment by paskal** [comment]
Thanks, will check and test shortly. Could you please add it to the v2 as well?

**Comment by paskal** [comment]
According to the rules:

> For your pull requests to be accepted, the request must either be merged, have the Hacktoberfest-accepted label, or have an overall approving review on GitHub.

I think this PR will be counted once merged.

**Comment by badolamgk** [comment]
Hey/Privyet @paskal 
I have done the required changes. Thanks for reviewing and providing reference for Remark42.

File: _example/main.go [hunk 1]
```diff
@@ -66,6 +66,9 @@ func main() {
 				if strings.HasPrefix(claims.User.ID, "patreon_") { // allow all users with ms auth
 					return true
 				}
+				if strings.HasPrefix(claims.User.ID, "discord_") { // allow all users with ms auth
```
Comment (paskal) [inline _example/main.go:69]: nitpick: comment here is copied wrongly

**Review by paskal — APPROVED** [review]

**Comment by badolamgk** [comment]
Hey @umputun

Let me know if there are any other changes required on my end. Cheers.

**Comment by badolamgk** [comment]
For the below command,
```shell
go test -timeout=60s -v -race -p 1 -covermode=atomic ./...
```
Test cases passing on my workstation, attaching logs [here](https://github.com/user-attachments/files/17267254/go-test-results.txt).

**Review by paskal — APPROVED** [review]

**Review by umputun — APPROVED** [review]

**Pull request merged**